src/vcSettingsUI.cpp

```diff
@@ -109,7 +109,14 @@ void vcSettingsUI_Show(vcState *pProgramState)
         {
           vcSettingsUI_ShowHeader(pProgramState, vcString::Get("settingsAppearance"), vcSC_Appearance);
 
-          vcSettingsUI_LangCombo(pProgramState);
+          bool languageWasJustChanged = false;
+          static bool languageWasChangedLastFrame = false;
+          vcSettingsUI_LangCombo(pProgramState, &languageWasJustChanged);
+          // Reset focus to the settings window (fixes a window ordering bug when language is changed for the first time)
+          if (languageWasChangedLastFrame)
+            ImGui::SetWindowFocus();
+          languageWasChangedLastFrame = languageWasJustChanged;
```

Comment:
This would be a bit cleaner:
```c++
static bool languageWasChanged = false;
if (languageWasChanged) // Needs to be off by one frame
  ImGui::SetWindowFocus();

vcSettingsUI_LangCombo(pProgramState, &languageWasChanged);
```

Reply:
Oops, forgot to assign back to false after the if.